### PR TITLE
Add linters #5

### LINKED ISSUE #5: Add linters
* Add golangci-lint as earthly target `+lint-go`
* Add buf lint as earthly target `+lint-proto`
* Add earthly target `+lint` that executes all lint targets - `+lint-go` and `+lint-proto`

### PR DESCRIPTION
**Pull request by VladislavsPerkanuks**

### CODE REVIEW

File: Earthfile [hunk 1]
```diff
@@ -9,3 +11,26 @@ proto:
   RUN --mount=type=cache,target=$BUF_CACHE_DIR buf build
   RUN --mount=type=cache,target=$BUF_CACHE_DIR buf generate
   SAVE ARTIFACT gen/proto/go/translate/v1 AS LOCAL pkg/server/translate/v1
+
+lint-go:
+  FROM golang:$go_version-alpine
+  RUN wget -O- -nv https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v$golangci_lint_version
+  RUN apk add build-base
```
Comment (VladislavsPerkanuks) [inline Earthfile]: Needed for golancgi-lint else error is thrown
https://github.com/golangci/golangci-lint/issues/3289

File: Earthfile [hunk 2]
```diff
@@ -9,3 +11,26 @@ proto:
...
+  WORKDIR /translate
+  COPY --dir pkg .
+  COPY .golangci.yml .
+  COPY go.mod go.sum .
+  RUN go mod download
```
Comment (VladislavsPerkanuks) [inline Earthfile:21]: Don't like this, as firstly: redownloads every time, if something is changes as we don't save the artifact, secondly: not really the thing that target should be doing.
My suggestion, another target that generates those files, go mod init, etc. and then saves as local artifact

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+run:
+  skip-dirs:
+    - pkg/server
```
Comment (janishorsts) [inline .golangci.yml:3]: Let's add the following lint rules

```yaml
linters:
  enable:
    - asciicheck
    - bidichk
    - bodyclose
    - containedctx
    - contextcheck
    - decorder
    - durationcheck
    - errcheck
    - errchkjson
    - errname
    - errorlint
    - execinquery
    - exhaustive
    - exportloopref
    - gocognit
    - goconst
    - gocritic
    - gocyclo
    - godot
    - gofumpt
    - gomnd
    - goprintffuncname
    - gosec
    - gosimple
    - govet
    - importas
    - ineffassign
    - lll
    - makezero
    - misspell
    - nakedret
    - nestif
    - nilerr
    - noctx
    - nolintlint
    - paralleltest
    - prealloc
    - staticcheck
    - typecheck
    - unconvert
    - unparam
    - unused
    - wrapcheck
    - wsl
linters-settings:
  govet:
    enable-all: true
```

File: Earthfile (hunk 1 above)
Comment (janishorsts) [reply Earthfile]: We are not gonna use cgo, you can disable it with `ENV CGO_ENABLED=0` or `RUN go env -w CGO_ENABLED=0`.

Comment (janishorsts) [inline Earthfile]:
```suggestion
  ENV CGO_ENABLED=0
```

File: Earthfile (hunk 2 above)
Comment (janishorsts) [reply Earthfile:21]: There are a few options with `earthly` to optimize it, the simplest one - you can create a target from another target (continue flow).

```earthly
deps:
  COPY go.mod go.sum .
  RUN go mod download
  SAVE ARTIFACT go.mod AS LOCAL go.mod
  SAVE ARTIFACT go.sum AS LOCAL go.sum
  
lint-go:
   FROM +deps
   RUN wget -O- -nv https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v$golangci_lint_version
   COPY --dir pkg .
   COPY .golangci.yaml .
   RUN golangci-lint run
```

File: Earthfile [hunk 4]
```diff
@@ -1,4 +1,6 @@
 VERSION 0.6
+ARG go_version=1.19.4
```
Comment (janishorsts) [inline Earthfile:2]: Be consistent in value assignment - in one place you don't have an assignment operator, in another you do.

```earthly
ARG go_version=1.19.4
// vs
ARG go_version 1.19.4
```

**Review by janishorsts — CHANGES_REQUESTED** [review]

**Review by janishorsts — APPROVED** [review]

**Pull request merged**